core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -3145,54 +3035,166 @@ protected function save($varValue)
 			}
 		}
 
+		// Make sure unique fields are unique
+		if ((\is_array($varValue) || (string) $varValue !== '') && ($arrData['eval']['unique'] ?? null) && !$this->Database->isUniqueValue($this->strTable, $this->strField, $varValue, $this->objActiveRecord->id))
+		{
+			throw new \Exception(sprintf($GLOBALS['TL_LANG']['ERR']['unique'], $arrData['label'][0] ?: $this->strField));
+		}
+
```

Comment:
Moving this is in fact also a behavior change that should be discussed in advance and – if we agree on it – be changed in all DC_ classes instead of only here.